ovn-org/ovn-kubernetes
# Automatically Set GOMAXPROCS for ovn-kube

**- What this PR does and why is it needed**


This PR uses `automaxprocs` to detect and set a sensible GOMAXPROCS value at runtime
from whatever limits the container runtime has set via cgroups. It also sets the CPU Limit to 1
for ovn-kube binaries such that we have an upper-bound limit.


**- Special notes for reviewers**


Changes:

- `cmd/*`
- `dist/templates`

**- How to verify it**


I'm going to see if I can get instrument a cluster to get some data on CPU throttling *before* and *after* by deploying this on a real cluster and applying some load.

**- Description for the changelog**
